### PR TITLE
append based Build()

### PR DESCRIPTION
**Pull request by alicebob**
Saves a few allocations.

### CODE REVIEW

File: command.go [hunk 1]
```diff
@@ -217,16 +216,16 @@ func resInt(x interface{}) (int, error) {
 }
 
 func buildCommand(fields []string) []byte {
-	var b bytes.Buffer
-	b.WriteByte('*')
-	b.WriteString(strconv.Itoa(len(fields)))
-	b.WriteString("\r\n")
+	b := make([]byte, 0, 64)
```
Comment (robx) [inline command.go]: I guess it's not worth it to estimate the size more precisely based on the fields.

**Comment by robx** [comment]
Looks good to me.

File: command.go [hunk 2]
```diff
@@ -217,16 +216,16 @@ func resInt(x interface{}) (int, error) {
...
 	for _, f := range fields {
-		b.WriteByte('$')
-		b.WriteString(strconv.Itoa(len(f)))
-		b.WriteString("\r\n")
-		b.WriteString(f)
-		b.WriteString("\r\n")
+		b = append(b, '$')
+		b = strconv.AppendInt(b, int64(len(f)), 10)
+		b = append(b, '\r', '\n')
+		b = append(b, []byte(f)...)
```
Comment (robx) [inline command.go]: As we were discussing, does this allocate? And if so, do we avoid it (by `append(b, f...)` or just looping over and indexing `f`?

Comment (alicebob) [inline command.go]: No difference in allocs, but does save a few ns per call.

**Pull request merged**